app/models/firm.rb

```diff
@@ -25,6 +25,7 @@ class Firm < ActiveRecord::Base
   belongs_to :parent, class_name: 'Firm'
 
   has_many :advisers, dependent: :destroy
+  has_many :offices, -> { order created_at: :asc }
```

Comment:
As offices are specific to firms, should we not be removing them i the firm is deleted?